common/BasicMultiQueueApplication.hpp

```diff
@@ -173,6 +173,14 @@ class BasicMultiQueueApplication : public virtual MonoDeviceApplication
 			else
 				m_transferDownQueue.qIx = familyQueueCounts[m_transferDownQueue.famIx]++;
 
+			if (!isHeadlessCompute())
+			{
+				queue_req_t graphicsQueueRequirement = { .requiredFlags = queue_flags_t::EQF_GRAPHICS_BIT,.disallowedFlags = queue_flags_t::EQF_NONE,.queueCount = 1 };
+				m_graphicsQueue.famIx = queueAllocator.allocateFamily(graphicsQueueRequirement, { queue_flags_t::EQF_NONE });
+				assert(m_graphicsQueue.famIx != QueueAllocator::InvalidIndex);
+				m_graphicsQueue.qIx = familyQueueCounts[m_graphicsQueue.famIx]++;
+			}
+
```

Comment:
this needs a better fix, you're right that I forgot to allocate the graphics queue.

The reason was because Graphics Queue Families need to be checked for compatibility with a swapchain.

For later examples I need to either rework the `allocateFamily` function to take a template lambda instead of `originalReq`, OR extend `SPhysicalDeviceFilter::QueueRequirement` to also include a list of surfaces the family must be able to present to.

Reply:
ok so about the better fix

If you notice, I have to handle allocation failures gracefully.

This is because old (non ARC) Intels only have one queue family that can provide only a single queue, therefore I have to alias failures.

So here I'd allocate the Graphics queue first, and assert it must have successfully allocated. Then if you fail to allocate a different queue for Compute, you should alias the compute queue to the graphics one.

Read the existing code see how it falls back and logs everything, follow similar convention.

You should also provide a list of unwanted flags for the Graphics queue allocation in order of most unwanted to least, so TRANSFER, SPARSE_BIND, COMPUTE, PROTECTED,